runtime/jvmti/jvmtiHook.c

```diff
@@ -570,13 +570,12 @@ jvmtiHookVMRestoreStartAgent(J9HookInterface **hook, UDATA eventNum, void *event
 {
 	J9VMThread *currentThread = ((J9RestoreEvent *)eventData)->currentThread;
 	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
```

Comment:
can you show me the call stack when this is called?

Reply:
> 578			criuRestoreStartAgent(vm);
```
#0  jvmtiHookVMRestoreStartAgent (hook=0x7ffff0012680, eventNum=110, eventData=0x7ffff7d36140, userData=0x7ffff00e7e20) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/jvmti/jvmtiHook.c:578
#1  0x00007ffff7f9b6ce in J9HookDispatch (hookInterface=0x7ffff0012680, taggedEventNum=110, eventData=0x7ffff7d36140) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/omr/util/hookable/hookable.cpp:237
#2  0x00007ffff7acf27d in criuCheckpointJVMImpl (env=0x1ff00, imagesDir=0x12ffe0, leaveRunning=0 '\000', shellJob=1 '\001', extUnixSupport=0 '\000', logLevel=0, logFile=0x0, fileLocks=1 '\001', workDir=0x12ffa8, tcpEstablished=1 '\001', autoDedup=0 '\000', 
    trackMemory=0 '\000', unprivileged=0 '\000', optionsFile=0x12ff80, environmentFile=0x0, ghostFileLimit=-1) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/CRIUHelpers.cpp:1940
#3  0x00007ffff5701848 in Java_openj9_internal_criu_InternalCRIUSupport_checkpointJVMImpl (env=0x1ff00, unused=0x192930, imagesDir=0x12ffe0, leaveRunning=0 '\000', shellJob=1 '\001', extUnixSupport=0 '\000', logLevel=0, logFile=0x0, fileLocks=1 '\001', 
    workDir=0x12ffa8, tcpEstablished=1 '\001', autoDedup=0 '\000', trackMemory=0 '\000', unprivileged=0 '\000', optionsFile=0x12ff80, environmentFile=0x0, ghostFileLimit=-1)
    at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/jcl/common/criu.cpp:125
#4  0x00007ffff7c0c58e in ffi_call_unix64 () at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/libffi/x86/unix64.S:105
#5  0x00007ffff7c0be0b in ffi_call_int (cif=0x7ffff7d36a30, fn=0x7ffff7f994e0, rvalue=0x20018, avalue=0x7ffff0672ec0, closure=0x0) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/libffi/x86/ffi64.c:672
#6  0x00007ffff7c0be8c in ffi_call (cif=0x7ffff7d36a30, fn=0x7ffff7f994e0, rvalue=0x20018, avalue=0x7ffff0672ec0) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/libffi/x86/ffi64.c:691
#7  0x00007ffff7b1d5b5 in VM_DebugBytecodeInterpreterCompressed::cJNICallout (this=0x7ffff7d37690, _sp=@0x7ffff7d375d8: 0x12ff40, _pc=@0x7ffff7d375e0: 0x7 <error: Cannot access memory at address 0x7>, receiverAddress=0x192930, javaArgs=0x12ffe8, 
    returnType=0x7ffff7d36ccb "", returnStorage=0x20018, function=0x7ffff7f994e0, isStatic=true) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/BytecodeInterpreter.hpp:2612
#8  0x00007ffff7b1cfb7 in VM_DebugBytecodeInterpreterCompressed::callCFunction (this=0x7ffff7d37690, _sp=@0x7ffff7d375d8: 0x12ff40, _pc=@0x7ffff7d375e0: 0x7 <error: Cannot access memory at address 0x7>, jniMethodStartAddress=0x7ffff7f994e0, receiverAddress=0x192930, 
    javaArgs=0x12ffe0, bp=0x7ffff7d36cd8, isStatic=true, returnType=0x7ffff7d36ccb "") at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/BytecodeInterpreter.hpp:2430
#9  0x00007ffff7b1c801 in VM_DebugBytecodeInterpreterCompressed::runJNINative (this=0x7ffff7d37690, _sp=@0x7ffff7d375d8: 0x12ff40, _pc=@0x7ffff7d375e0: 0x7 <error: Cannot access memory at address 0x7>)
    at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/BytecodeInterpreter.hpp:2318
#10 0x00007ffff7b3a707 in VM_DebugBytecodeInterpreterCompressed::run (this=0x7ffff7d37690, vmThread=0x1ff00) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/BytecodeInterpreter.hpp:10970
#11 0x00007ffff7b16a11 in debugBytecodeLoopCompressed (currentThread=0x1ff00) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/BytecodeInterpreter.inc:112
#12 0x00007ffff7b67362 in c_cInterpreter () at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-slowdebug/vm/runtime/vm/xcinterp.s:157
#13 0x00007ffff7a1e892 in runCallInMethod (env=0x1ff00, receiver=0x0, clazz=0x130170, methodID=0x7ffff05ae038, args=0x7ffff7d37ca8) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/callin.cpp:1174
#14 0x00007ffff7a58147 in gpProtectedRunCallInMethod (entryArg=0x7ffff7d37c40) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/jnicsup.cpp:300
#15 0x00007ffff7b7a385 in signalProtectAndRunGlue (portLibrary=0x7ffff7cf6080 <j9portLibrary>, userData=0x7ffff7d37bf0) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/util/jniprotect.c:45
#16 0x00007ffff788d7fb in omrsig_protect (portLibrary=0x7ffff7cf6080 <j9portLibrary>, fn=0x7ffff7b7a355 <signalProtectAndRunGlue>, fn_arg=0x7ffff7d37bf0, handler=0x7ffff7a4fe5f <structuredSignalHandler>, handler_arg=0x1ff00, flags=506, result=0x7ffff7d37bd8)
    at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/omr/port/unix/omrsignal.c:425
#17 0x00007ffff7b7a4f1 in gpProtectAndRun (function=0x7ffff7a580e6 <gpProtectedRunCallInMethod(void*)>, env=0x1ff00, args=0x7ffff7d37c40) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/util/jniprotect.c:78
#18 0x00007ffff7a58703 in gpCheckCallin (env=0x1ff00, receiver=0x0, cls=0x130170, methodID=0x7ffff05ae038, args=0x7ffff7d37ca8) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/jnicsup.cpp:488
#19 0x00007ffff7a573f3 in callStaticVoidMethod (env=0x1ff00, cls=0x130170, methodID=0x7ffff05ae038) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/openj9/runtime/vm/jnicgen.c:384
#20 0x00007ffff7fa9036 in JavaMain (_args=0x7fffffff9f60) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/src/java.base/share/native/libjli/java.c:622
#21 0x00007ffff7fb05a8 in ThreadJavaMain (args=0x7fffffff9f60) at /root/builds/jdk11-criuassertV2-1114/openj9-openjdk-jdk11/src/java.base/unix/native/libjli/java_md_solinux.c:790
#22 0x00007ffff7e04ac3 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#23 0x00007ffff7e96850 in ?? () from /lib/x86_64-linux-gnu/libc.so.6
```